src/coreclr/jit/inlinepolicy.cpp

```diff
@@ -1802,6 +1810,41 @@ double ExtendedDefaultPolicy::DetermineMultiplier()
         JITDUMP("\nCallsite is in a no-return region.  Multiplier limited to %g.", multiplier);
     }
 
+    if (m_HasPinvokes)
+    {
+        // We have pinvoke call(s) in the inlinee. It means that if we inline the callee we'll inject a lot of code
+        // into the caller's prolog/epilog (SuppressGCTransition pinvokes won't do it, but those are rare), unless
+        // it already contains pinvokes.
+
+        // So, first of all, let's avoid inlining methods with pinvokes to cold sections
+        // we don't want to introduce overhead in the caller's prolog for rarely executed pinvokes.
+        if (m_HasProfileWeights && m_ProfileFrequency < 0.2)
```

Comment:
> m_ProfileFrequency < 0.2

Can you break this down for me? I assume this means less than 20 % of the time? Seems like 20 % should be documented somewhere or at least clarified as to why that number was chosen. It is also possible I'm ignorant of how the JIT computes weights and the reason is obvious to the JIT team. If so, please forgive my ignorance.

Reply:
Sort of, it means that the current block, we inline the given callee to, has weigth=0.2 (you can read it as Probability to be executed as 20% compared to the probability of the very first block of the caller = 100%)

This function to calculate the benefit multiplier is full of magic constants we just set based on some benchmarks/experimetns. Initially, I planned to limit the m_ProfileFrequency to be 0.0 (cold/rarely taken), but then decided to slightly increase it based on a few apps I tested it. The logic is basically (let's inline callees with pinvokes only when they're guaratneed to be on the hot path). The value is dicussable.